File: DSA/Ayush_Mishra_590014844/Day_20_Question_1.py
def sort_students_by_grade(students):
    return sorted(students, key=lambda x: x[1], reverse=True)

n = int(input("Enter number of students: "))
students = []
for _ in range(n):
    name = input("Enter student name: ")
    grade = int(input("Enter grade: "))
    students.append((name, grade))

sorted_students = sort_students_by_grade(students)
print("Sorted students by grade (descending):")
for student in sorted_students:
    print(student)

File: DSA/Ayush_Mishra_590014844/Day_20_Question_1.c
#include <stdio.h>
#include <string.h>

typedef struct {
    char name[50];
    int grade;
} Student;

void sortStudents(Student arr[], int n) {
    for(int i = 0; i < n-1; i++)
        for(int j = i+1; j < n; j++)
            if(arr[i].grade < arr[j].grade) {
                Student temp = arr[i];
                arr[i] = arr[j];
                arr[j] = temp;
            }
}

int main() {
    int n;
    printf("Enter number of students: ");
    scanf("%d", &n);
    Student students[n];

    for(int i = 0; i < n; i++) {
        printf("Enter name and grade of student %d: ", i+1);
        scanf("%s %d", students[i].name, &students[i].grade);
    }

    sortStudents(students, n);

    printf("Sorted students by grade (descending):\n");
    for(int i = 0; i < n; i++)
        printf("%s %d\n", students[i].name, students[i].grade);

    return 0;
}
